config.sh

```diff
@@ -275,7 +275,7 @@ WIN_PERL=${WIN_PERL:-"C:/perl/bin/perl.exe"}
 # Android default settings and tools
 if test "${OS}" = "android" ; then
     ANDROID_NDK_VERSION=${ANDROID_NDK_VERSION:-r10d}
-    ANDROID_PLATFORM=${ANDROID_PLATFORM:-android-10}
+    ANDROID_PLATFORM=${ANDROID_PLATFORM:-android-17}
```

Comment:
The instructions we have for installing the Android toolchain install all SDKs, rather than just API 10, so anyone who followed the instructions should have it. If not, it is fairly easy to install manually (just run the Android helper tool in the SDK and select the API17 SDK bundle to install).

As far as I can see, this is the correct way to support older platforms while getting access to new APIs where available on Android. The alternative would be using the Java introspection stuff to see if the methods are present at runtime...